aioredis/commands/hash.py

```diff
@@ -50,12 +54,34 @@ def hmget(self, key, field, *fields, encoding=_NOTSET):
         return self._conn.execute(b'HMGET', key, field, *fields,
                                   encoding=encoding)
 
-    # TODO: replace args with dict_or_pairs
-    def hmset(self, key, field, value, *pairs):
+    def hmset(self, key, *args, **kwargs):
         """Set multiple hash fields to multiple values."""
-        if len(pairs) % 2 != 0:
-            raise TypeError("length of pairs must be even number")
-        return wait_ok(self._conn.execute(b'HMSET', key, field, value, *pairs))
+        if args:
+            is_dict = tuple(map(lambda x: isinstance(x, dict), args))
+        elif not kwargs:
+            raise TypeError("length of args or kwargs must be > 0")
+
+        if not args:
+            pairs = functools.reduce(operator.add, kwargs.items())
+
+        elif all(is_dict):
```

Comment:
why do you need all `args` to be dicts?
what is the reason for calling hmset like this:
```python
redis.hmset(key, {'a': 1}, {'b': 2})
```
and not simply with one dict?